airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -246,9 +278,32 @@ def submit(self, application="", **kwargs):
                 )
             )
 
-    def _process_log(self, itr):
+        self.log.debug("Should track driver: {}".format(self._should_track_driver_status))
+
+        # We want the Airflow job to wait until the Spark driver is finished
+        if self._should_track_driver_status:
+            if self._driver_id is None:
+                raise AirflowException(
+                    "No driver id is known: something went wrong when executing the spark submit command"
+                )
+
+            # We start with the SUBMITTED status as initial status
+            self._driver_status = "SUBMITTED"
+
+            # Start tracking the driver status (blocking function)
+            self._start_driver_status_tracking()
+
+            if self._driver_status != "FINISHED":
+                raise AirflowException(
+                    "ERROR : Driver {} badly exited with status {}".format(self._driver_id, self._driver_status)
+                )
+
+    def _process_spark_submit_log(self, itr):
         """
-        Processes the log files and extracts useful information out of it
+        Processes the log files and extracts useful information out of it.
+
+        Remark: If the driver needs to be tracked for its status, the log-level of the spark deploy needs to be
+        at least INFO (log4j.logger.org.apache.spark.deploy.rest=INFO)
```

Comment:
I updated the log4j again the other day, I now think it's  even better to log like this: 
```
log4j.logger.org.apache.spark.deploy=INFO
```